### PR TITLE
Added 3 new links to watch for cheating and updated warning system to keep track of how many links are shared. 

### PR DESCRIPTION
**Pull request by serspencer**
Links now include

1. Chegg
2. Coursehero
3. Quizlet

### CODE REVIEW

**Comment by serspencer** [comment]
Added reporting feature to link sharing and it will issue a warning to the link sender.

**Comment by serspencer** [comment]
Updated to now issue a warning when links are shared and logs the user.

**Comment by serspencer** [comment]
Removed extra import.

File: src/app/handlers/blacklist.handler.ts [hunk 1]
```diff
@@ -1,9 +1,16 @@
 import { TextChannel } from 'discord.js';
 import Constants from '../../common/constants';
-import { IContainer, IHandler, IMessage, ClassType } from '../../common/types';
-
+import { IContainer, IHandler, IMessage, ClassType, linkLabel } from '../../common/types';
```
Comment (c650) [inline src/app/handlers/blacklist.handler.ts:3]: Capitalise `linkLabel`. Also I doubt it'll be used outside this file (blacklist.handler), so might as well just define the type here.

File: src/app/handlers/blacklist.handler.ts [hunk 2]
```diff
@@ -15,9 +22,15 @@ export class BlacklistHandler implements IHandler {
...
-      if (message.content.toLowerCase().match(expression)) {
+    this._expressions.forEach(({regex, label}) => {
+      if (message.content.toLowerCase().match(regex)) {
         this.container.messageService.sendBotReport(message);
+        const rep = new Moderation.Report(
+          message.guild,
+          message.author.username,
+          `Please do not post ${label} links.`,
+        );
+        this.container.modService.fileReport(rep);
```
Comment (c650) [inline src/app/handlers/blacklist.handler.ts:33]: I thought you also wanted to notify the user. This won't do that. It only files a report.

File: src/common/types.ts [hunk 3]
```diff
@@ -124,3 +124,8 @@ export interface ILoggerWrapper {
   input(message: any, ...args: any[]): any;
   silly(message: any, ...args: any[]): any;
 }
+
+export interface linkLabel {
+  regex: RegExp;
+  label: string;
+}
```
Comment (c650) [inline src/common/types.ts:131]: Needs final \n

But also see above comment about moving this definition into blacklist.handler

**Review by c650 — CHANGES_REQUESTED** [review]

**Comment by serspencer** [comment]
LabelLink is capital, user is now DM'ed and code has been prettied up.

File: src/app/handlers/blacklist.handler.ts [hunk 4]
```diff
@@ -22,13 +27,16 @@ export class BlacklistHandler implements IHandler {
       return;
     }
 
-    this._expressions.forEach(({regex, label}) => {
+    this._expressions.forEach(({ regex, label }) => {
       if (message.content.toLowerCase().match(regex)) {
+        message.author.send(
+          `Please do not share \`${label}\` links in the \`${message.guild.name}\` server.`
```
Comment (c650) [inline src/app/handlers/blacklist.handler.ts:33]: use regular `'` to avoid escaping

File: src/app/handlers/blacklist.handler.ts [hunk 5]
```diff
@@ -22,13 +27,16 @@ export class BlacklistHandler implements IHandler {
...
+        );
         this.container.messageService.sendBotReport(message);
         const rep = new Moderation.Report(
           message.guild,
           message.author.username,
-          `Please do not post ${label} links.`,
+          `${message.author.username} shared a ${label} link.`
```
Comment (c650) [inline src/app/handlers/blacklist.handler.ts:39]: no need to include username in the description bcs its saved as part of report already

File: .gitignore [hunk 6]
```diff
@@ -3,3 +3,4 @@ dist/**/*
 src/config/config.json
 .env
 *.log
+settings.json
```
Comment (c650) [inline .gitignore:6]: ?

**Review by c650 — CHANGES_REQUESTED** [review]
Tests?

File: .gitignore (hunk 6 above)
Comment (serspencer) [reply .gitignore:6]: It wanted to add my VScode settings to the GitHub so its being ignored.

File: src/app/handlers/blacklist.handler.ts (hunk 4 above)
Comment (serspencer) [reply src/app/handlers/blacklist.handler.ts:33]: The back ticks were there for formatting.  Not sure if that is what you are referring too.

File: .gitignore (hunk 6 above)
Comment (c650) [reply .gitignore:6]: hmm. I don't want to taint the gitignore too much. Can you make those vsc settings global so there's no need for a settings file in the workspace?

File: src/app/handlers/blacklist.handler.ts (hunk 4 above)
Comment (c650) [reply src/app/handlers/blacklist.handler.ts:33]: nvm, im confused

**Comment by serspencer** [comment]
Updated as per requests

**Review by c650 — APPROVED** [review]

**Pull request merged**